File: src/main/java/nz/co/searchwellington/filters/PageParameterFilter.java
package nz.co.searchwellington.filters;

import javax.servlet.http.HttpServletRequest;

import org.springframework.context.annotation.Scope;
import org.springframework.stereotype.Component;

@Component
@Scope("request")
public class PageParameterFilter implements RequestAttributeFilter {

	public final static String PAGE_ATTRIBUTE = "page";

	public void filter(HttpServletRequest request) {
		if (request.getParameter(PAGE_ATTRIBUTE) != null) {
			String pageString = request.getParameter(PAGE_ATTRIBUTE);
			try {
				Integer page = Integer.parseInt(pageString);
				request.setAttribute(PAGE_ATTRIBUTE, page);
			} catch (NumberFormatException e) {
			}
		}
	}

}

File: src/main/scala/nz/co/searchwellington/filters/attributesetters/PageParameterFilter.scala
package nz.co.searchwellington.filters.attributesetters

import jakarta.servlet.http.HttpServletRequest
import org.springframework.stereotype.Component

import scala.concurrent.Future

@Component
class PageParameterFilter extends AttributeSetter {

  override def setAttributes(request: HttpServletRequest): Future[Map[String, Any]] = {
    if (request.getParameter(PageParameterFilter.PAGE_ATTRIBUTE) != null) {
      val pageString = request.getParameter(PageParameterFilter.PAGE_ATTRIBUTE)
      try {
        Future.successful(Map(
          PageParameterFilter.PAGE_ATTRIBUTE -> pageString.toInt
        ))
      } catch {
        case e: NumberFormatException =>
          Future.successful(Map.empty)
      }
    } else {
      Future.successful(Map.empty)
    }
  }
}

object PageParameterFilter {
  val PAGE_ATTRIBUTE = "page"
}
